src/net_processing.cpp

```diff
@@ -579,13 +579,13 @@ struct CNodeState {
       */
     struct ChainSyncTimeoutState {
         //! A timeout used for checking whether our peer has sufficiently synced
-        int64_t m_timeout;
+        int64_t m_timeout{0};
         //! A header with the work we require on our peer's chain
-        const CBlockIndex * m_work_header;
+        const CBlockIndex* m_work_header{nullptr};
         //! After timeout is reached, set to true after sending getheaders
-        bool m_sent_getheaders;
+        bool m_sent_getheaders{false};
         //! Whether this peer is protected from disconnection due to a bad/slow chain
-        bool m_protect;
+        bool m_protect{false};
```

Comment:
`#21424` is partial because some changes can be applied only when #19109 is done.